### ISSUE TITLE
Upgrading redux-toolkit from 1.2.5 gave stack trace on saga put()

### ISSUE THREAD

**Issue by steinarb** [reduxjs/redux-toolkit#671]
When I tried upgrading redux-toolkit from 1.2.5 to 1.4.0 today, I got the following stack trace:
```
Error: [Immer] minified error nr: 19 MapSet. Find the full error at: https://bit.ly/3cXEKWf
    at zt (immer.esm.js:1)
    at en (immer.esm.js:1)
    at yn (immer.esm.js:1)
    at e.t.produce (immer.esm.js:1)
    at redux-toolkit.esm.js:882
    at Array.reduce (<anonymous>)
    at redux-toolkit.esm.js:853
    at redux.js:445
    at p (<anonymous>:1:36402)
    at v (<anonymous>:1:36684)
```

Immer error 19 referred to in the stack trace, is "plugin not loaded"

The error occurs already at 1.3.0 and downgrading redux-toolkit in package.json doesn't fix the problem.

If I restore the package-lock.json file from before upgrading redux-toolkit from 1.2.5, the problem goes away.

**Comment by markerikson** [comment]
Any chance you have multiple versions of Immer in your app after an upgrade?

**Comment by steinarb** [comment]
>>>>> Mark Erikson <notifications@github.com>:

> Any chance you have multiple versions of Immer in your app after an upgrade?

Yes, I have two versions of Immer, but I had that both before and after the upgrade.

The other version of Immer is 1.10 and the dependecy comes from
react-dev-utils 10.2.1.

Before the upgrade @reduxjs/toolkit pulled in immer 4.0.2.

After bumping @reduxjs/toolkit to 1.4.0 and rebuilding, @reduxjs/toolkit
pulled in immer 7.0.7

react-dev-utils pulled in immer 1.10 in both cases (I don't even know
where the stuff that pulls in react-dev-utils resides, because it's
nothing in the app outside of package-lock.json).

Thanks!

**Comment by markerikson** [comment]
Yeah, I'd filed an issue against CRA a while back that their version of Immer was wayyyy out of date:

https://github.com/facebook/create-react-app/issues/8750

That just got closed and I would assume the fix should be out in the next release (CRA 3.4.2?)

Does use of RTK 1.4 + Immer 7 fix the issue, or is that still happening?

**Comment by steinarb** [comment]
>>>>> Mark Erikson <notifications@github.com>:

> Yeah, I'd filed an issue against CRA a while back that their version of Immer was wayyyy out of date:
> https://github.com/facebook/create-react-app/issues/8750

> That just got closed and I would assume the fix should be out in the next release (CRA 3.4.2?)

Ok, thanks for the info and update.

> Does use of RTK 1.4 + Immer 7 fix the issue, or is that still happening?

I tried explictly depending on immer 7 like this, in package.json:

    "dependencies": {
        ...
        "@reduxjs/toolkit": "^1.4.0",
        "immer": "^7.0.7",
        "terser": "^4.8.0"
    },

Unfortunately, that didn't help.

I also tried using npm-force-resolutions, to force immer to version 7,
but I was unable to get npx to work with frontend-maven-plugin, which is
what I use to create and deliver bundle.js.

So, I guess I'm stuck with RTK 1.2.5, until CRA fixes their immer
dependency.

Thanks!

**Comment by markerikson** [comment]
I know that Yarn has a `resolutions` feature as well.  Not sure if switching to Yarn would make a difference.

Since there doesn't seem to be anything else actionable on our end, I'll go ahead and close this.
